packages/core/src/storages/sitemap_request_list.ts

```diff
@@ -390,6 +417,10 @@ export class SitemapRequestList implements IRequestList {
         await requestList.restoreState();
         void requestList.load({ parseSitemapOptions: options.parseSitemapOptions });
 
+        if (requestList.persistenceOptions.enable) {
+            requestList.events.on(EventType.PERSIST_STATE, requestList.persistState);
```

Comment:
Won't this possibly change the `this` binding of `RequestList.persistState()` to the `EventManager` instance? The way to go here is IMO `() => requestList.persistState()` or `requestList.persistState.bind(requestList)`